From b7ad7adb04a27f7771645064bf50f89aac6f26be Mon Sep 17 00:00:00 2001
From: Andy Yan <andy.yan@rock-chips.com>
Date: Wed, 24 Feb 2021 10:28:58 +0800
Subject: [PATCH] Revert "drm/rockchip: framebuffer: add rockchip_fbdev_blank"

This reverts commit 2d36048f1c1f038938dc8e3f012dbf8c0e320d3c.
Which causes some confused behavior in suspend and resume.

Change-Id: Ibff257aea6bcecc9a012397d83a9d8a1ffd02132
Signed-off-by: Andy Yan <andy.yan@rock-chips.com>
---
 drivers/gpu/drm/rockchip/rockchip_drm_fbdev.c | 11 +----------
 1 file changed, 1 insertion(+), 10 deletions(-)

diff --git a/drivers/gpu/drm/rockchip/rockchip_drm_fbdev.c b/drivers/gpu/drm/rockchip/rockchip_drm_fbdev.c
index d8e080fae71a..e349106bdaa6 100644
--- a/drivers/gpu/drm/rockchip/rockchip_drm_fbdev.c
+++ b/drivers/gpu/drm/rockchip/rockchip_drm_fbdev.c
@@ -50,20 +50,12 @@ static struct dma_buf *rockchip_fbdev_get_dma_buf(struct fb_info *info)
 	return buf;
 }
 
-static int rockchip_fbdev_blank(int blank, struct fb_info *info)
-{
-	struct drm_fb_helper *helper = info->par;
-
-	drm_fb_helper_restore_fbdev_mode_unlocked(helper);
-
-	return drm_fb_helper_blank(blank, info);
-}
-
 static struct fb_ops rockchip_drm_fbdev_ops = {
 	.owner		= THIS_MODULE,
 	.fb_check_var	= drm_fb_helper_check_var,
 	.fb_set_par	= drm_fb_helper_set_par,
 	.fb_setcmap	= drm_fb_helper_setcmap,
+	.fb_blank	= drm_fb_helper_blank,
 	.fb_pan_display	= drm_fb_helper_pan_display,
 	.fb_debug_enter = drm_fb_helper_debug_enter,
 	.fb_debug_leave = drm_fb_helper_debug_leave,
@@ -74,7 +66,6 @@ static struct fb_ops rockchip_drm_fbdev_ops = {
 	.fb_read	= drm_fb_helper_sys_read,
 	.fb_write	= drm_fb_helper_sys_write,
 	.fb_mmap		= rockchip_fbdev_mmap,
-	.fb_blank		= rockchip_fbdev_blank,
 	.fb_dmabuf_export	= rockchip_fbdev_get_dma_buf,
 };
 
-- 
2.35.3

